### PR TITLE
Migration to webpacker & client folder removal

### PR DESCRIPTION
**Pull request by Punkbooster**
Should be merged after: https://github.com/spark-solutions/spark-starter-kit/pull/161

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          "browsers": "> 1%",
+          "uglify": true
+        },
+        "useBuiltIns": true
+      }
+    ],
+    "react"
+  ],
+  "plugins": [
+    "syntax-dynamic-import",
```
Comment (piotrleniec-spark) [inline .babelrc]: Do we really need these plugins if we use `env` preset?

File: Procfile.dev [hunk 2]
```diff
@@ -1,15 +1 @@
-# Basic procfile for dev work.
```
Comment (piotrleniec-spark) [inline Procfile.dev]: This file isn't needed, we can remove it.

File: app/javascript/packs/application.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/* eslint no-console:0 */
+// This file is automatically compiled by Webpack, along with any other files
+// present in this directory. You're encouraged to place your actual application logic in
+// a relevant structure within app/javascript and only use these pack files to reference
+// that code so it'll be compiled.
+//
+// To reference this file, add <%= javascript_pack_tag 'application' %> to the appropriate
+// layout file, like app/views/layouts/application.html.erb
+
+console.log('Hello World from Webpacker')
```
Comment (piotrleniec-spark) [inline app/javascript/packs/application.js]: This `console.log` isn't needed.

File: app/javascript/packs/application.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+// Support component names which are placed under containers alias folder name
+const componentRequireContext = require.context('providers', true)
+const ReactRailsUJS = require('react_ujs')
```
Comment (piotrleniec-spark) [inline app/javascript/packs/application.js]: Let's use `import` instead of `require`.

File: app/javascript/packs/constants/index.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import mirrorCreator from 'mirror-creator'
```
Comment (piotrleniec-spark) [inline app/javascript/packs/constants/index.js]: 👍

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -7,5 +7,10 @@
   <body class='<%= "#{controller_name}" + " " + "#{controller_name}" + "-" + "#{action_name}" %>'>
     <%= render 'spree/shared/google_analytics.js' %>
     <%= yield %>
+
+    <%= react_component 'Store', { sampleReducer: { name: 'Default Name' } }, { prerender: false } %>
```
Comment (piotrleniec-spark) [inline app/views/layouts/application.html.erb]: Let's set `prerender` to `true`.

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -7,5 +7,10 @@
...
+    <%= react_component 'HelloWorldProvider', {}, { prerender: false } %>
```
Comment (piotrleniec-spark) [inline app/views/layouts/application.html.erb]: Let's set `prerender` to `true`.

**Comment by Punkbooster** [comment]
Should be merged after: https://github.com/spark-solutions/spark-starter-kit/pull/161

File: app/javascript/packs/providers/HelloWorldProvider.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { Provider } from 'react-redux'
+import { getStore } from 'providers/Store'
```
Comment (michalprzybysz) [inline app/javascript/packs/providers/HelloWorldProvider.js]: I think you can use it this way: `import store, { getStore } from 'providers/Store'

**Pull request closed without merging**